astropy/units/tests/test_quantity.py

```diff
@@ -1055,12 +1055,31 @@ def test_quantity_pickelability():
     assert q1.unit == q2.unit
 
 
-def test_quantity_from_string():
-    with pytest.raises(TypeError):
-        q = u.Quantity(u.m * "5")
-        # the reverse should also fail once #1408 is in
+def test_quantity_initialisation_from_unit():
+    q = u.Quantity(u.m)
+    assert q.unit == u.m
+    assert q.value == 1.
+    assert u.Unit(q) == u.m
+    q = u.Quantity('m')
```

Comment:
The reason for one and not zero, and for my ambivalence, is because of this:
```
In [14]: (1 * u.km) / u.m
Out[14]: <Quantity 1.0 km / m>
```
In other words in unit arithmetic a bare `u.m` is precisely equivalent to `1.0 * u.m`.  From a practical standpoint they are always the same, so it becomes a pedantic question about whether we want to allow that.